GameDataParser/Parsers/QuestParser.cs

```diff
@@ -0,0 +1,260 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.MemoryMappedFiles;
+using System.Xml;
+using GameDataParser.Crypto.Common;
+using GameDataParser.Files;
+using Maple2Storage.Types.Metadata;
+using ProtoBuf;
+
+namespace GameDataParser.Parsers
+{
+    public static class QuestParser
+    {
+        public static List<QuestMetadata> Parse(MemoryMappedFile m2dFile, IEnumerable<PackFileEntry> entries)
+        {
+            List<QuestMetadata> quests = new List<QuestMetadata>();
+            foreach (PackFileEntry entry in entries)
+            {
+
+                if (!entry.Name.StartsWith("quest/"))
+                {
+                    continue;
+                }
+
+                QuestMetadata metadata = new QuestMetadata();
+
+                using XmlReader reader = m2dFile.GetReader(entry.FileHeader);
+                while (reader.Read())
+                {
+                    if (reader.NodeType != XmlNodeType.Element)
+                    {
+                        continue;
+                    }
+
+                    if (reader.Name == "environment" && (reader["locale"] != "NA" && reader["locale"] != "")) // only reading quests for NA or all environments
+                    {
+                        reader.Skip();
+                    }
+
+                    if (reader.Name == "environment")
+                    {
+                        metadata.Feature = reader["feature"];
+                        metadata.Locale = reader["locale"];
+                    }
+
+                    if (reader.Name == "basic")
```

Comment:
Why not `else if`?

Reply:
No reason, if you want I can change to `else if`

Reply:
Well this code as it stands will keep checking for different values even after it might have already been established earlier, so yeah